pkg/infrastructure/azure/azure.go

```diff
@@ -393,13 +407,26 @@ func (p *Provider) PostProvision(ctx context.Context, in clusterapi.PostProvisio
 		return fmt.Errorf("error retrieving Azure session: %w", err)
 	}
 	subscriptionID := ssn.Credentials.SubscriptionID
+	cloudConfiguration := ssn.CloudConfig
 
 	if in.InstallConfig.Config.Publish == types.ExternalPublishingStrategy {
-		vmClient, err := armcompute.NewVirtualMachinesClient(subscriptionID, ssn.TokenCreds, nil)
+		vmClient, err := armcompute.NewVirtualMachinesClient(subscriptionID, ssn.TokenCreds,
+			&arm.ClientOptions{
+				ClientOptions: policy.ClientOptions{
+					Cloud: cloudConfiguration,
```

Comment:
```suggestion
					Cloud: ssn.CloudConfig,
```

Reply:
Which common pattern? Why save the value to a local var if it's more clear you're just using a value from the session by doing `session.CloudConfig`?